src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -783,348 +157,156 @@ const std::vector<impl_desc_type>& FullyConnected::getDefaultImplPriority() {
+              elementsCount,
+              ", zerosCount = ",
+              zerosCount,
+              ", nnzCount = ",
+              elementsCount - zerosCount);
 
-        for (size_t i = 0; i < descInputNumbers(); i++) {
-            auto desc = getSrcMemDesc(prim_desc, i);
-            const auto inputBlockedMask = (supportsUndefStridesAndOffset() && !(i == WEIGHTS_ID && useSparseWeights)) ?
-                BlockedMemoryDesc::EMPTY_MASK :
-                BlockedMemoryDesc::FULL_MASK;
+    auto sparseRate = static_cast<float>(zerosCount) / static_cast<float>(elementsCount);
 
-            inConfs.emplace_back(desc, inputBlockedMask);
-        }
+    DEBUG_LOG("Sparse rate = ",
+              sparseRate * 100,
+              "%, min sparse rate = ",
+              minSparseRate * 100,
+              "%, use sparse weights = ",
+              sparseRate >= minSparseRate);
 
-        const auto outputBlockedMask = supportsUndefStridesAndOffset() ? BlockedMemoryDesc::EMPTY_MASK : BlockedMemoryDesc::FULL_MASK;
-
-        for (size_t i = 0; i < descOutputNumbers(); i++) {
-            auto desc = getDstMemDesc(prim_desc, i);
-
-            outConfs.emplace_back(desc, outputBlockedMask, inPlaceOutPort);
-        }
-
-        const NodeConfig config(inConfs, outConfs);
-        const impl_desc_type impl_type = parse_impl_name(prim_desc.impl_info_str());
+    return sparseRate < minSparseRate;
+}
 
-        supportedPrimitiveDescriptors.emplace_back(config, impl_type);
+void FullyConnected::initSupportedPrimitiveDescriptors() {
+    attrs.withBias = getOriginalInputsNumber() == 3;
+    attrs.dequantizationScales = getDQScales();
+    attrs.sparseWeights = useSparseWeightsDecompression(getParentEdgeAt(DATA_ID)->getParent(),
+                                                        getOriginalInputPrecisionAtPort(DATA_ID),
+                                                        context->getConfig().fcSparseWeiDecompressionRate);
+    postOps = getPostOps(fusedWith);
+
+    const auto& srcTypes = getOriginalInputPrecisions();
+    auto dstTypes = getOriginalOutputPrecisions();
+    // @todo graph optimizer should update original output precisions instead
+    if (!fusedWith.empty())
+        dstTypes = fusedWith.back()->getOriginalOutputPrecisions();
+
+    VecMemoryDescs srcDescs;
+    const auto& creatorsMap = BlockedDescCreator::getCommonCreators();
+    for (size_t i = 0; i < srcTypes.size(); i++) {
+        const auto srcDesc = creatorsMap.at(LayoutType::ncsp)->createSharedDesc(srcTypes[i], getInputShapeAtPort(i));
+        srcDescs.push_back(srcDesc);
+    }
+
+    VecMemoryDescs dstDescs;
+    for (size_t i = 0; i < dstTypes.size(); i++) {
+        const auto dstDesc = creatorsMap.at(LayoutType::ncsp)->createSharedDesc(dstTypes[i], getOutputShapeAtPort(i));
+        dstDescs.push_back(dstDesc);
+    }
+
+    auto createEmptyMemoryDesc = [](const ov::element::Type type) {
+        return std::make_shared<CpuBlockedMemoryDesc>(type, Shape{0});
```

Comment:
`Shape{0}` actually represents a zero tensor of first rank, thus this is a legit memory descriptor, which in general case doesn't necessary means "empty" memory. Though it has been already discussed in https://github.com/openvinotoolkit/openvino/pull/20718#discussion_r1457681439, but I decided to post this comment as the final solution is still absent.

Reply:
Will be properly implemented in scope of the following commit

Reply:
Added to leftovers